spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -144,6 +134,7 @@ public void destroyTask(@PathVariable("name") String name) {
 			throw new NoSuchTaskDefinitionException(name);
 		}
 		repository.delete(name);
+		taskLauncher.destroy(name);
```

Comment:
Should the `definition` repository `delete` happen after the completion of `destroy`?